SOAPBindingTest is failing with NoClassDefFoundError - SOAPException

### Describe the bug

SOAPBindingTest is failing with following exception:

```
2022-04-02T13:36:15.9438231Z [ERROR] Tests run: 10, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 3.457 s <<< FAILURE! - in org.keycloak.testsuite.saml.SOAPBindingTest
2022-04-02T13:36:15.9438948Z -- [ERROR] org.keycloak.testsuite.saml.SOAPBindingTest.ecpFlowCreatesTransientSessions  Time elapsed: 0.292 s  <<< ERROR!
2022-04-02T13:36:15.9439684Z -- org.keycloak.testsuite.runonserver.RunOnServerException: java.lang.NoClassDefFoundError: javax/xml/soap/SOAPException
2022-04-02T13:36:15.9440377Z -- 	at org.keycloak.testsuite.client.KeycloakTestingClient$Server.run(KeycloakTestingClient.java:155)
2022-04-02T13:36:15.9441085Z -- 	at org.keycloak.testsuite.saml.SOAPBindingTest.ecpFlowCreatesTransientSessions(SOAPBindingTest.java:285)
2022-04-02T13:36:15.9441755Z -- 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-04-02T13:36:15.9442368Z -- 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-04-02T13:36:15.9443051Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9443600Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9444106Z -- 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
2022-04-02T13:36:15.9444674Z -- 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-04-02T13:36:15.9445271Z -- 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
2022-04-02T13:36:15.9445908Z -- 	at org.jboss.arquillian.junit.Arquillian$8$1.invokeMethod(Arquillian.java:325)
2022-04-02T13:36:15.9446412Z -- 	at org.jboss.arquillian.junit.MethodInvoker$1.invoke(MethodInvoker.java:18)
2022-04-02T13:36:15.9447058Z -- 	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
2022-04-02T13:36:15.9447734Z -- 	at org.keycloak.testsuite.arquillian.ModelTestExecutor.execute(ModelTestExecutor.java:49)
2022-04-02T13:36:15.9448304Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor433.invoke(Unknown Source)
2022-04-02T13:36:15.9448933Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9449470Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9449958Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9450554Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
2022-04-02T13:36:15.9451174Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
2022-04-02T13:36:15.9451717Z -- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
2022-04-02T13:36:15.9452299Z -- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
2022-04-02T13:36:15.9452802Z -- 	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
2022-04-02T13:36:15.9453446Z -- 	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:50)
2022-04-02T13:36:15.9454085Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor432.invoke(Unknown Source)
2022-04-02T13:36:15.9454713Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9455254Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9455734Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9456334Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
2022-04-02T13:36:15.9456950Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
2022-04-02T13:36:15.9457707Z -- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
2022-04-02T13:36:15.9458598Z -- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
2022-04-02T13:36:15.9459309Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor431.invoke(Unknown Source)
2022-04-02T13:36:15.9459928Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9460464Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9460948Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9461524Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9462170Z -- 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
2022-04-02T13:36:15.9462768Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor97.invoke(Unknown Source)
2022-04-02T13:36:15.9463389Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9463926Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9464407Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9464981Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9465622Z -- 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
2022-04-02T13:36:15.9466281Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor45.invoke(Unknown Source)
2022-04-02T13:36:15.9466910Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9467453Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9467931Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9468501Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9469142Z -- 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
2022-04-02T13:36:15.9469737Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor44.invoke(Unknown Source)
2022-04-02T13:36:15.9470361Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9470902Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9471373Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9471931Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9472525Z -- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
2022-04-02T13:36:15.9473130Z -- 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
2022-04-02T13:36:15.9473706Z -- 	at org.jboss.arquillian.junit.MethodInvoker.invoke(MethodInvoker.java:15)
2022-04-02T13:36:15.9474210Z -- 	at org.jboss.arquillian.junit.Arquillian$8.evaluate(Arquillian.java:332)
2022-04-02T13:36:15.9474693Z -- 	at org.jboss.arquillian.junit.Arquillian$4.evaluate(Arquillian.java:204)
2022-04-02T13:36:15.9475182Z -- 	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
2022-04-02T13:36:15.9475838Z -- 	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
2022-04-02T13:36:15.9476331Z -- 	at org.jboss.arquillian.junit.Arquillian$5.evaluate(Arquillian.java:215)
2022-04-02T13:36:15.9476793Z -- 	at org.jboss.arquillian.junit.Arquillian$7$1.invoke(Arquillian.java:279)
2022-04-02T13:36:15.9487957Z -- 	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.execute(ClientBeforeAfterLifecycleEventExecuter.java:88)
2022-04-02T13:36:15.9489044Z -- 	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.on(ClientBeforeAfterLifecycleEventExecuter.java:66)
2022-04-02T13:36:15.9489818Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor153.invoke(Unknown Source)
2022-04-02T13:36:15.9490456Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9491019Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9491536Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9492152Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
2022-04-02T13:36:15.9495492Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
2022-04-02T13:36:15.9496258Z -- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
2022-04-02T13:36:15.9497157Z -- 	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createBeforeContext(ContainerEventController.java:114)
2022-04-02T13:36:15.9497861Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor151.invoke(Unknown Source)
2022-04-02T13:36:15.9498490Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9499037Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9499528Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9500314Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9501025Z -- 	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
2022-04-02T13:36:15.9501676Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor97.invoke(Unknown Source)
2022-04-02T13:36:15.9502365Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9502966Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9503509Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9504127Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9504847Z -- 	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
2022-04-02T13:36:15.9505453Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor45.invoke(Unknown Source)
2022-04-02T13:36:15.9506084Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9506642Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9507200Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9507768Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9508418Z -- 	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
2022-04-02T13:36:15.9509016Z -- 	at jdk.internal.reflect.GeneratedMethodAccessor44.invoke(Unknown Source)
2022-04-02T13:36:15.9509647Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9510183Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9510668Z -- 	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
2022-04-02T13:36:15.9511230Z -- 	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
2022-04-02T13:36:15.9511778Z -- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
2022-04-02T13:36:15.9512296Z -- 	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
2022-04-02T13:36:15.9512938Z -- 	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.fireCustomLifecycle(EventTestRunnerAdaptor.java:159)
2022-04-02T13:36:15.9513543Z -- 	at org.jboss.arquillian.junit.Arquillian$7.evaluate(Arquillian.java:273)
2022-04-02T13:36:15.9514085Z -- 	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
2022-04-02T13:36:15.9514596Z -- 	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
2022-04-02T13:36:15.9515137Z -- 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
2022-04-02T13:36:15.9515936Z -- 	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
2022-04-02T13:36:15.9516458Z -- 	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
2022-04-02T13:36:15.9516911Z -- 	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
2022-04-02T13:36:15.9517399Z -- 	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
2022-04-02T13:36:15.9517882Z -- 	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
2022-04-02T13:36:15.9518350Z -- 	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
2022-04-02T13:36:15.9518822Z -- 	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:166)
2022-04-02T13:36:15.9519342Z -- 	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
2022-04-02T13:36:15.9519863Z -- 	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
2022-04-02T13:36:15.9520672Z -- 	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:177)
2022-04-02T13:36:15.9521155Z -- 	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
2022-04-02T13:36:15.9521714Z -- 	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
2022-04-02T13:36:15.9522177Z -- 	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:115)
2022-04-02T13:36:15.9522729Z -- 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:364)
2022-04-02T13:36:15.9523359Z -- 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
2022-04-02T13:36:15.9523984Z -- 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:237)
2022-04-02T13:36:15.9524573Z -- 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:158)
2022-04-02T13:36:15.9525186Z -- 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
2022-04-02T13:36:15.9525780Z -- 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2022-04-02T13:36:15.9526325Z -- 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
2022-04-02T13:36:15.9526861Z -- 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
2022-04-02T13:36:15.9527356Z -- Caused by: java.lang.NoClassDefFoundError: javax/xml/soap/SOAPException
2022-04-02T13:36:15.9527883Z -- 	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
2022-04-02T13:36:15.9528365Z -- 	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
2022-04-02T13:36:15.9528827Z -- 	at java.base/java.lang.Class.getDeclaredMethod(Class.java:2473)
2022-04-02T13:36:15.9529302Z -- 	at java.base/java.lang.invoke.SerializedLambda$1.run(SerializedLambda.java:231)
2022-04-02T13:36:15.9529800Z -- 	at java.base/java.lang.invoke.SerializedLambda$1.run(SerializedLambda.java:228)
2022-04-02T13:36:15.9530289Z -- 	at java.base/java.security.AccessController.doPrivileged(Native Method)
2022-04-02T13:36:15.9530812Z -- 	at java.base/java.lang.invoke.SerializedLambda.readResolve(SerializedLambda.java:228)
2022-04-02T13:36:15.9531375Z -- 	at java.base/jdk.internal.reflect.GeneratedMethodAccessor319.invoke(Unknown Source)
2022-04-02T13:36:15.9532009Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9532559Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9533050Z -- 	at java.base/java.io.ObjectStreamClass.invokeReadResolve(ObjectStreamClass.java:1265)
2022-04-02T13:36:15.9533604Z -- 	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2231)
2022-04-02T13:36:15.9534133Z -- 	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1681)
2022-04-02T13:36:15.9534642Z -- 	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:493)
2022-04-02T13:36:15.9535135Z -- 	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:451)
2022-04-02T13:36:15.9535937Z -- 	at org.keycloak.testsuite.integration-arquillian-testsuite-providers@999-SNAPSHOT//org.keycloak.testsuite.runonserver.SerializationUtil.decode(SerializationUtil.java:40)
2022-04-02T13:36:15.9537006Z -- 	at org.keycloak.testsuite.integration-arquillian-testsuite-providers@999-SNAPSHOT//org.keycloak.testsuite.rest.TestingResourceProvider.runOnServer(TestingResourceProvider.java:810)
2022-04-02T13:36:15.9537792Z -- 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-04-02T13:36:15.9538403Z -- 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-04-02T13:36:15.9539069Z -- 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-04-02T13:36:15.9539607Z -- 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-04-02T13:36:15.9540215Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
2022-04-02T13:36:15.9540941Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
2022-04-02T13:36:15.9541789Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
2022-04-02T13:36:15.9542673Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
2022-04-02T13:36:15.9543518Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
2022-04-02T13:36:15.9544423Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
2022-04-02T13:36:15.9545326Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
2022-04-02T13:36:15.9546102Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
2022-04-02T13:36:15.9546943Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
2022-04-02T13:36:15.9547746Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
2022-04-02T13:36:15.9548491Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
2022-04-02T13:36:15.9549229Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
2022-04-02T13:36:15.9549989Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
2022-04-02T13:36:15.9550772Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
2022-04-02T13:36:15.9551689Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
2022-04-02T13:36:15.9552565Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
2022-04-02T13:36:15.9553320Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
2022-04-02T13:36:15.9554158Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:249)
2022-04-02T13:36:15.9555075Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:60)
2022-04-02T13:36:15.9556041Z -- 	at org.jboss.resteasy.resteasy-core@4.7.4.Final//org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
2022-04-02T13:36:15.9556732Z -- 	at javax.servlet.api@2.0.0.Final//javax.servlet.http.HttpServlet.service(HttpServlet.java:590)
2022-04-02T13:36:15.9557383Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
2022-04-02T13:36:15.9558092Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
2022-04-02T13:36:15.9558897Z -- 	at org.keycloak.keycloak-wildfly-extensions@999-SNAPSHOT//org.keycloak.provider.wildfly.WildFlyRequestFilter.lambda$doFilter$0(WildFlyRequestFilter.java:41)
2022-04-02T13:36:15.9559737Z -- 	at org.keycloak.keycloak-services@999-SNAPSHOT//org.keycloak.services.filters.AbstractRequestFilter.filter(AbstractRequestFilter.java:43)
2022-04-02T13:36:15.9560636Z -- 	at org.keycloak.keycloak-wildfly-extensions@999-SNAPSHOT//org.keycloak.provider.wildfly.WildFlyRequestFilter.doFilter(WildFlyRequestFilter.java:39)
2022-04-02T13:36:15.9561354Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
2022-04-02T13:36:15.9562022Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
2022-04-02T13:36:15.9562707Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
2022-04-02T13:36:15.9563528Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
2022-04-02T13:36:15.9564289Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
2022-04-02T13:36:15.9565040Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
2022-04-02T13:36:15.9566025Z -- 	at org.wildfly.security.elytron-web.undertow-server@1.10.1.Final//org.wildfly.elytron.web.undertow.server.ElytronRunAsHandler.lambda$handleRequest$1(ElytronRunAsHandler.java:68)
2022-04-02T13:36:15.9566994Z -- 	at org.wildfly.security.elytron-base@1.18.3.Final//org.wildfly.security.auth.server.FlexibleIdentityAssociation.runAsFunctionEx(FlexibleIdentityAssociation.java:103)
2022-04-02T13:36:15.9567814Z -- 	at org.wildfly.security.elytron-base@1.18.3.Final//org.wildfly.security.auth.server.Scoped.runAsFunctionEx(Scoped.java:161)
2022-04-02T13:36:15.9568485Z -- 	at org.wildfly.security.elytron-base@1.18.3.Final//org.wildfly.security.auth.server.Scoped.runAs(Scoped.java:73)
2022-04-02T13:36:15.9569312Z -- 	at org.wildfly.security.elytron-web.undertow-server@1.10.1.Final//org.wildfly.elytron.web.undertow.server.ElytronRunAsHandler.handleRequest(ElytronRunAsHandler.java:67)
2022-04-02T13:36:15.9570156Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
2022-04-02T13:36:15.9571029Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:117)
2022-04-02T13:36:15.9577439Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
2022-04-02T13:36:15.9578757Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
2022-04-02T13:36:15.9579591Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
2022-04-02T13:36:15.9580603Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
2022-04-02T13:36:15.9581907Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
2022-04-02T13:36:15.9582955Z -- 	at org.wildfly.security.elytron-web.undertow-server-servlet@1.10.1.Final//org.wildfly.elytron.web.undertow.server.servlet.CleanUpHandler.handleRequest(CleanUpHandler.java:38)
2022-04-02T13:36:15.9584198Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
2022-04-02T13:36:15.9585059Z -- 	at org.wildfly.extension.undertow@26.0.1.Final//org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
2022-04-02T13:36:15.9585884Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
2022-04-02T13:36:15.9586960Z -- 	at org.wildfly.extension.undertow@26.0.1.Final//org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler.handleRequest(GlobalRequestControllerHandler.java:68)
2022-04-02T13:36:15.9588318Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.SendErrorPageHandler.handleRequest(SendErrorPageHandler.java:52)
2022-04-02T13:36:15.9589051Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
2022-04-02T13:36:15.9589810Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:275)
2022-04-02T13:36:15.9590584Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:79)
2022-04-02T13:36:15.9591285Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:134)
2022-04-02T13:36:15.9591982Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:131)
2022-04-02T13:36:15.9592853Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
2022-04-02T13:36:15.9593674Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
2022-04-02T13:36:15.9594630Z -- 	at org.wildfly.extension.undertow@26.0.1.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1544)
2022-04-02T13:36:15.9595791Z -- 	at org.wildfly.extension.undertow@26.0.1.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1544)
2022-04-02T13:36:15.9596854Z -- 	at org.wildfly.extension.undertow@26.0.1.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1544)
2022-04-02T13:36:15.9597878Z -- 	at org.wildfly.extension.undertow@26.0.1.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1544)
2022-04-02T13:36:15.9598791Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:255)
2022-04-02T13:36:15.9599545Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:79)
2022-04-02T13:36:15.9600275Z -- 	at io.undertow.servlet@2.2.14.Final//io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:100)
2022-04-02T13:36:15.9600913Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.server.Connectors.executeRootHandler(Connectors.java:387)
2022-04-02T13:36:15.9601533Z -- 	at io.undertow.core@2.2.14.Final//io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:852)
2022-04-02T13:36:15.9602230Z -- 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
2022-04-02T13:36:15.9602945Z -- 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
2022-04-02T13:36:15.9603799Z -- 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
2022-04-02T13:36:15.9604484Z -- 	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
2022-04-02T13:36:15.9605079Z -- 	at org.jboss.xnio@3.8.5.Final//org.xnio.XnioWorker$WorkerThreadFactory$1$1.run(XnioWorker.java:1280)
2022-04-02T13:36:15.9605532Z -- 	at java.base/java.lang.Thread.run(Thread.java:829)
2022-04-02T13:36:15.9606067Z -- Caused by: java.lang.ClassNotFoundException: javax.xml.soap.SOAPException
2022-04-02T13:36:15.9606579Z -- 	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
2022-04-02T13:36:15.9607062Z -- 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
2022-04-02T13:36:15.9607530Z -- 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
2022-04-02T13:36:15.9607858Z -- 	... 93 more
```

The reason for this exception is, that when RunOnServer is used within this test class, the class is loaded in the container JVM. For some reason, SOAPException cannot be found there. Simple way of fixing this is not using the class in the test class

### Version

17.0.1

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

--- a/saml-core/src/test/java/org/keycloak/saml/common/util/StaxParserUtilTest.java
+++ b/saml-core/src/test/java/org/keycloak/saml/common/util/StaxParserUtilTest.java
@@ -22,7 +22,6 @@ import org.keycloak.saml.processing.core.parsers.util.HasQName;
 
 import java.nio.charset.Charset;
 import java.util.NoSuchElementException;
-import java.util.function.BiFunction;
 
 import javax.xml.stream.XMLEventReader;
 import javax.xml.stream.XMLStreamException;
@@ -249,9 +248,13 @@ public class StaxParserUtilTest {
         }
     }
 
+    private interface StartElementHasQNameBooleanBiFunction {
+        Boolean apply(StartElement el, HasQName qName);
+    }
+
     @Test
     public void testGetBooleanAttributeValue() throws XMLStreamException, ParsingException {
-        testGetBooleanAttributeValue(new BiFunction<StartElement, HasQName, Boolean>() {
+        testGetBooleanAttributeValue(new StartElementHasQNameBooleanBiFunction() {
             @Override
             public Boolean apply(StartElement t, HasQName u) {
                 return StaxParserUtil.getBooleanAttributeValue(t, u);
@@ -261,7 +264,7 @@ public class StaxParserUtilTest {
 
     @Test
     public void testGetBooleanAttributeValueRP() throws XMLStreamException, ParsingException {
-        testGetBooleanAttributeValue(new BiFunction<StartElement, HasQName, Boolean>() {
+        testGetBooleanAttributeValue(new StartElementHasQNameBooleanBiFunction() {
             @Override
             public Boolean apply(StartElement t, HasQName u) {
                 return StaxParserUtil.getBooleanAttributeValueRP(t, u);
@@ -269,7 +272,7 @@ public class StaxParserUtilTest {
         });
     }
 
-    private void testGetBooleanAttributeValue(BiFunction<StartElement, HasQName, Boolean> predicate) throws XMLStreamException, ParsingException {
+    private void testGetBooleanAttributeValue(StartElementHasQNameBooleanBiFunction predicate) throws XMLStreamException, ParsingException {
         testGetBooleanAttributeValue("<a AllowCreate=\"false\">text</a>", predicate, false);
         testGetBooleanAttributeValue("<a AllowCreate=\"true\">text</a>", predicate, true);
         testGetBooleanAttributeValue("<a AllowCreate=\"0\">text</a>", predicate, false);
@@ -278,7 +281,7 @@ public class StaxParserUtilTest {
         testGetBooleanAttributeValue("<a>text</a>", predicate, null);
     }
 
-    private void testGetBooleanAttributeValue(String xml, BiFunction<StartElement, HasQName, Boolean> predicate, Boolean expectedResult) throws XMLStreamException, ParsingException {
+    private void testGetBooleanAttributeValue(String xml, StartElementHasQNameBooleanBiFunction predicate, Boolean expectedResult) throws XMLStreamException, ParsingException {
         XMLEventReader reader = StaxParserUtil.getXMLEventReader(IOUtils.toInputStream(xml, Charset.defaultCharset()));
 
         assertThat(reader.nextEvent(), instanceOf(StartDocument.class));

